File: leetcode/easy/26. Remove Duplicates from Sorted Array/sample.php
<?php
class Solution {

    /**
     * @param Integer[] $nums
     * @return Integer
     */
    function removeDuplicates(&$nums) {
        // solution #1
        $nums = array_unique($nums);
        return count($nums);

        // solution #2
        $numsLenght = count($nums);
        $lastUniqueElement = $nums[0];

        for ($i = 1; $i < $numsLenght; $i++) {
            if ($nums[$i] !== $lastUniqueElement) $lastUniqueElement = $nums[$i];
            else unset($nums[$i]);
        }
        return count($nums);
    }
}

class Solution {
    /**
     * @param Integer[] $nums
     * @return Integer
     */
    function removeDuplicates(&$nums) {
        $count = count($nums);
        $writeIndex = 1;

        for ($i = 1; $i < $count; $i++) {
            if ($nums[$i] !== $nums[$i - 1]) {
                $nums[$writeIndex] = $nums[$i];
                $writeIndex++;
            }
        }

        return $writeIndex;
    }
}



?>

File: leetcode/easy/26. Remove Duplicates from Sorted Array/sample.py
class Solution(object):
    def removeDuplicates(self, nums):
        replace = 1
        for i in range(1, len(nums)):
            # replaceを配列の2個目にセット
            # 2個目以降から実施。すぐ後ろの要素と比較していくreplaceをずらす
            if nums[i-1] != nums[i]:
                nums[replace] = nums[i]
                replace += 1
        return replace